src/worker.js

```diff
@@ -250,6 +280,21 @@ worker.emit = function (payload) {
   }
 };
 
+/**
+ * Builds an api for attaching to a method for
+ * worker functionality availble from within a task context
+ * @returns {Object}
+ */
+function _buildMethodWorkerApi() {
```

Comment:
I think instead of a function `_buildMethodWorkerApi`, we can create a new static object named for example `publicWorker` or so, alongside the `worker` object, and let it have the public methods `addAbortListener` and `emit`, since they will not change over time.

So we get something along the lines of:

```js
const worker = { ... }

const publicWorker = { 
  addAbortListener: worker.addAbortListener,
  emit: worker.emit
}

Reply:
Are we able to make this static? since at time of definition `worker` will not have `emit` defined? I think it might be fine since `worker` is the root object and will be referenced passed.